snekbox/api/resources/eval.py

```diff
@@ -123,10 +125,21 @@ def on_post(self, req: falcon.Request, resp: falcon.Response) -> None:
         if "input" in body:
             body.setdefault("args", ["-c"])
             body["args"].append(body["input"])
+
+        binary_path = body.get("binary_path")
+        if binary_path:
+            binary_path = Path(binary_path)
+            if (
+                not binary_path.resolve().as_posix().startswith("/lang/")
```

Comment:
Oh I missed this in your description

> if supplied, snekbox ensures that the path is within /lang (to avoid executing arbitrary binaries on the fs) and that the file exists.

It is fairly trivial to bypass this restriction by using the `execve` syscall. We cannot block this syscall either (at least not through nsjail's seccomp filters). That is due to a quirk in which the filter is applied before nsjail's own `execve` call, causing nsjail to commit suicide.

Granted, there may be use cases which don't have binaries in `/lang` that make it possible to call syscalls. But I think if a request comes up to have such restriction, we can add it as a later feature.